apps/web/src/pages/templates/WorkflowListPage.tsx

```diff
@@ -156,15 +157,18 @@ function WorkflowListPage() {
   const { loading: areNotificationGroupLoading } = useNotificationGroup();
   const {
     templates,
-    loading,
+    loading: areWorkflowsLoading,
+    isFetching,
     totalItemCount,
     totalPageCount,
     currentPageNumber,
+    pageSize,
+    search,
+    setSearch,
     setCurrentPageNumber,
     setPageSize,
-    pageSize,
   } = useTemplates({ areSearchParamsEnabled: true });
-  const isLoading = areNotificationGroupLoading || loading;
+  const [searchValue, setSearchValue] = useState(search ?? '');
```

Comment:
the search value state for the input to reflect quickly what user types; the request, and URL updates are debounced